README.md

```diff
@@ -1,11 +1,272 @@
 
-In Modomoto we came up with the idea of __COSS - Component Oriented Style Sheets__ framework. 
+In MODOMOTO we came up with the idea of __COSS - Component Oriented Style Sheets__ framework. The idea of Component Oriented CSS is not new, but in here we focus on the practical application of it.
+
+__COSS__ is a way of rethinking, refactoring and building our CSS code base using components. The basic idea is to wrap parts of our HTML into components with a meaningful name as the component, style them in a separate CSS file that has the same name of the component and that only contains styles declarations for it (including media queries if necessary).
+
+COSS involves changes in our CSS, HTML and file structure and it requires SASS or LESS.
+
+## PHILOSOPHY
+
+The CSS must be preprocessed (SASS, LESS etc);
+
+A component is a div with only one class and the class name is the name of the component;
+
+The HTML should be split into re-usable components;
+
+All the CSS rules of a component should be defined into a component file;
+
+Pixels, Color Codes and Fonts Families should be defined by a variable with a meaningful name;
+
+It is not allowed to define Pixels, Color Codes and Fonts Families in a component, variables/mixins should be used instead;
+
+The basic structure of files in COSS is ```assets/stylesheets/components```, ```assets/stylesheets/variables/```, ```assets/stylesheets/mixins/``` and ```assets/stylesheets/base.sass``` (where the shared CSS rules of the project are defined);
+
+When forcing the use of components, variables and mixins it is easier to make the css more consistent;
+
+
+## HTML
+  In every HTML document should be possible to identify part of code that are can be small 'components'. A component can be a 'payment form', an 'header menu', a 'login sidebar', a 'pennant' etc. Each of them should be identified with a self explanatory name.
+
+  The guideslines for creating HTML components are:
+
+  1. The name of the component should indicate what the component does, how does it look like to easily identify them when browsing the code;
+  2. Possibily the component should have be reusable, so should be the name;
+  3. The name of it should not be too generic.
+
+
+  ```HTML
+  <!-- good -->
+  <div class'sidebar_login'>
+  </div>
+
+  <!-- bad -->
+  <div class'sidebar_login'>
+  </div>
+
+  <!-- goodish -->
+  <div class'sidebar_login_homepage'>
+  </div>
+
+  ```
+
+## FILE STRUCTURE
+
+  Components are found only in a folder ```assets/stylesheets/components```
+
+  After defining our new component in our HTML, we can create a new file in the ```assets/stylesheets/components``` that has exactly the same name of the component. For instance if we take the component ```sidebar_login``` , we should have a file named ```assets/stylesheets/components/sidebar_login.sass```
```

Comment:
general question who do we address? Do we describe this guide in collaborative terms (After defining our component), frontal (After defining your component) or neutral (After defining a component in the HTML)?